src/System.IO.FileSystem/tests/Enumeration/TrimmedPaths.netcoreapp.cs

```diff
@@ -47,18 +47,183 @@ public void TrimmedPathsDeletion_Windows()
+                }
+
+                using (StreamWriter writer = fi.CreateText())
+                {
+                    writer.Write("foo");
+                }
+
+                using (StreamReader reader = fi.OpenText())
+                {
+                    Assert.Equal("foo", reader.ReadToEnd());
+                }
+
+                using (StreamWriter writer = fi.AppendText())
+                {
+                    writer.Write("bar");
+                }
+
+                using (StreamReader reader = fi.OpenText())
+                {
+                    Assert.Equal("foobar", reader.ReadToEnd());
+                }
+            }
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void TrimmedPathsCopyTo_Windows()
+        {
+            // Trailing spaces and periods are eaten when normalizing in Windows, making them impossible
+            // to access without using the \\?\ device syntax. We should, however, be able to copy them 
+            // without the special syntax from the info class when enumerating.
+
+            DirectoryInfo directory = Directory.CreateDirectory(GetTestFilePath());
+            string fileOne = Path.Join(directory.FullName, "Trailing space ");
+            string fileTwo = Path.Join(directory.FullName, "Trailing period.");
+            File.Create(@"\\?\" + fileOne).Dispose();
+            File.Create(@"\\?\" + fileTwo).Dispose();
+
+            FileInfo[] files = directory.GetFiles();
+            Assert.Equal(2, files.Length);
+            foreach (FileInfo fi in directory.GetFiles())
+            {
+                FileInfo newInfo = fi.CopyTo(Path.Join(directory.FullName, GetTestFileName()));
+                Assert.True(newInfo.Exists);
+                FileInfo newerInfo = fi.CopyTo(Path.Join(directory.FullName, GetTestFileName()), overwrite: true);
+                Assert.True(newerInfo.Exists);
+            }
+
+            Assert.Equal(6, directory.GetFiles().Length);
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void TrimmedPathsReplace_Windows()
+        {
+            // Trailing spaces and periods are eaten when normalizing in Windows, making them impossible
+            // to access without using the \\?\ device syntax. We should, however, be able to replace them
+            // from the info class when enumerating.
+
+            DirectoryInfo directory = Directory.CreateDirectory(GetTestFilePath());
+            string fileOne = Path.Join(directory.FullName, "Trailing space ");
+            string fileTwo = Path.Join(directory.FullName, "Trailing period.");
+            File.WriteAllText(@"\\?\" + fileOne, "space");
+            File.WriteAllText(@"\\?\" + fileTwo, "period");
+
+            FileInfo[] files = directory.GetFiles();
+            Assert.Equal(2, files.Length);
+
+            FileInfo destination = new FileInfo(Path.Join(directory.FullName, GetTestFilePath()));
```

Comment:
Maybe add "DeleteTestFilePath" to FileCleanupTestBase? Dispose can call it.

Reply:
Not sure what you're suggesting here.

Reply:
Disregard.